cg/meta/upload/scout/raredisease_config_builder.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import re
+from pathlib import Path
+
+from housekeeper.store.models import Version
+
+from cg.apps.lims import LimsAPI
+from cg.apps.madeline.api import MadelineAPI
+from cg.constants.housekeeper_tags import HK_DELIVERY_REPORT_TAG
+from cg.constants.scout import (
+    RAREDISEASE_CASE_TAGS,
+    RAREDISEASE_SAMPLE_TAGS,
+    UploadTrack,
+)
+from cg.constants.subject import RelationshipStatus
+from cg.meta.upload.scout.hk_tags import CaseTags, SampleTags
+from cg.meta.upload.scout.scout_config_builder import ScoutConfigBuilder
+from cg.meta.workflow.raredisease import RarediseaseAnalysisAPI
+from cg.models.scout.scout_load_config import (
+    CaseImages,
+    CustomImages,
+    Eklipse,
+    RarediseaseLoadConfig,
+    ScoutIndividual,
+    ScoutLoadConfig,
+    ScoutRarediseaseIndividual,
+)
+from cg.store.models import Analysis, Case, CaseSample, Sample
+
+LOG = logging.getLogger(__name__)
+
+
+class RarediseaseConfigBuilder(ScoutConfigBuilder):
+    def __init__(
+        self,
+        hk_version_obj: Version,
+        analysis_obj: Analysis,
+        raredisease_analysis_api: RarediseaseAnalysisAPI,
+        lims_api: LimsAPI,
+        madeline_api: MadelineAPI,
+    ):
+        super().__init__(
+            hk_version_obj=hk_version_obj,
+            analysis_obj=analysis_obj,
+            lims_api=lims_api,
+        )
+        self.case_tags: CaseTags = CaseTags(**RAREDISEASE_CASE_TAGS)
+        self.sample_tags: SampleTags = SampleTags(**RAREDISEASE_SAMPLE_TAGS)
+        self.load_config: RarediseaseLoadConfig = RarediseaseLoadConfig(
+            track=UploadTrack.RARE_DISEASE.value,
+            delivery_report=self.get_file_from_hk({HK_DELIVERY_REPORT_TAG}),
+        )
+        self.raredisease_analysis_api: RarediseaseAnalysisAPI = raredisease_analysis_api
+        self.lims_api: LimsAPI = lims_api
+        self.madeline_api: MadelineAPI = madeline_api
+
+    def build_load_config(self) -> RarediseaseLoadConfig:
+        """Create a RAREDISEASE specific load config for uploading analysis to Scout."""
+        LOG.info("Build load config for RAREDISEASE case")
+        self.load_config = RarediseaseLoadConfig()
+        self.load_config = self.add_common_info_to_load_config()
+        self.load_config.gene_panels = self.raredisease_analysis_api.get_aggregated_panels(
+            customer_id=self.analysis_obj.case.customer.internal_id,
+            default_panels=set(self.analysis_obj.case.panels),
+        )
+        self.include_case_files()
+        self.load_config = self.get_sample_information(load_config=self.load_config)
+        self.load_config = self.include_pedigree_picture()
+        self.load_config.custom_images = self.load_custom_image_sample()
```

Comment:
I think instantiating the object and returning it made this method a lot clearer. To improve it further I would pass the `RarediseaseLoadConfig` object to each method that interacts with it. An example line could look like:
`self.add_common_info_to_load_config(load_config)`.

Assigning the config to `self.load_config` and then using that to access the object within the methods makes it hard to keep track of the config. Sort of like a shell game...

Reply:
now all the methods are passed a load_config and return one for all workflows

Reply:
Sorry I think my comment might have been unclear. I think passing the load_config has clarified a lot, but I think we can rely on in-place modification instead of reassigning load_config every time.

Ie `add_common_info_to_load_config` and all the other methods operating on the load_config does not need to return it at the end. Let me know if you want a clarification 😊

Reply:
Do you mean something like this then?
```
    def add_common_info_to_load_config(self, load_config: ScoutLoadConfig):
        """Add the mandatory common information to a Scout load config object."""
        self.load_config = load_config
        self.load_config.analysis_date = self.analysis_obj.completed_at or self.analysis_obj.started_at
        self.load_config.default_gene_panels = self.analysis_obj.case.panels
        self.load_config.family = self.analysis_obj.case.internal_id
        self.load_config.family_name = self.analysis_obj.case.name
        self.load_config.owner = self.analysis_obj.case.customer.internal_id
        self.load_config.synopsis = self.analysis_obj.case.synopsis
        self.include_cohorts(load_config=self.load_config)
        self.include_phenotype_groups(load_config=self.load_config)
        self.include_phenotype_terms(load_config=self.load_config)
```

Reply:
Or leave `load_config` unused?